src/imu/sensors/magnetometer.cpp

```diff
@@ -0,0 +1,100 @@
+#include "magnetometer.hpp"
+
+namespace imu {
+
+auto Magnetometer::Init(const std::uint8_t i2c_address) noexcept -> types::DriverStatus {
+  SENSOR_DATA_REGISTER = imu::MAGNETOMETER_MEASUREMENT_DATA;
+  REGISTER_DATA_LENGTH_IN_BYTES = 7;
+
+  if (SensorGeneral::Init(i2c_address) != types::DriverStatus::OK)
+    return types::DriverStatus::HAL_ERROR;
+
+  SetSensorValues(0, 0, 0);
+  SetInitData();
+
+  initialized_ = true;
+  return types::DriverStatus::OK;
+}
+
+auto Magnetometer::SetInitData(void) noexcept -> void {
+  PowerDownMagnetometer();
+  EnterFuseROMAccessMode();
+  GetCalibrationValues();
+  PowerDownMagnetometer();
+  ConfigureForContinuousRead();
+}
+
+auto Magnetometer::PowerDownMagnetometer(void) noexcept -> void {
+  WriteContentIntoRegister(AK8963_CNTL, 0x00);
+  utilities::Sleep(10);
+}
+
+auto Magnetometer::EnterFuseROMAccessMode(void) noexcept -> void {
+  WriteContentIntoRegister(AK8963_CNTL, 0x0F);
+  utilities::Sleep(10);
+}
+
+auto Magnetometer::ConfigureForContinuousRead(void) noexcept -> void {
+  // Configure the magnetometer for continuous read and highest resolution.
+  // Set Bit 4 to enable 16 bit resolution and enable continuous mode data acquisition at 8 Hz.
+  // See MPU-9255 Register Map, Revision 1.0, p. 51
+  WriteContentIntoRegister(AK8963_CNTL, 0b10010);
```

Comment:
0b10010 could be a variable named continuous_read_configuration, then it would not be a magic number

Reply:
Magic number removed.